src/MessagePack/MessagePackUnityOptions.cs

```diff
@@ -0,0 +1,13 @@
+// Copyright (c) All contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace MessagePack;
+
+public static class MessagePackUnityOptions
+{
+    public static bool IL2CPP { get; set; }
```

Comment:
Wouldn't it be better to consider the environments which disallow dynamic-code-generation instead of focusing only on IL2CPP?

[RuntimeFeature.IsDynamicCodeSupported](https://learn.microsoft.com/en-us/dotnet/api/system.runtime.compilerservices.runtimefeature.isdynamiccodesupported?view=net-8.0#system-runtime-compilerservices-runtimefeature-isdynamiccodesupported) can be used since netstandard2.1.